app/controllers/reviews_controller.rb

```diff
@@ -2,13 +2,48 @@ class ReviewsController < ApplicationController
   def new
     @task = API::Task.includes(:framework).find(params[:task_id]).first
 
-    submission = API::Submission.includes(:entries).find(params[:submission_id]).first
+    @invoices_count = submission_entries.count { |e| e.source['sheet'].match?(/invoice/i) }
+    @orders_count = submission_entries.count { |e| e.source['sheet'].match?(/order/i) }
+
+    @validator = Validator.new(submission_entries: submission_entries)
 
-    @invoices_count = submission.entries.count { |e| e.source['sheet'].match?(/invoice/i) }
-    @orders_count = submission.entries.count { |e| e.source['sheet'].match?(/order/i) }
+    case @validator.status
+    when 'validated'
+      flash.now[:notice] = 'All entries are valid'
+    when 'errored'
+      flash.now[:alert] = "There are #{@validator.errors.count} errors in the uploaded spreadsheet"
+    end
   end
 
   def create
-    # NO OP
+    render 'tasks/complete'
+  end
+
+  def processing
+    redirect_to new_task_submission_review_path if ingest_entries_completed?
```

Comment:
This redirect always happens - at the point where the file has just been uploaded, the number of rows is still `nil` and there are no entries (`Validator` incorrect assumes there are no errors)